common/api/core-backend.api.md

```diff
@@ -2177,6 +2174,9 @@ export class GeometryPart extends DefinitionElement {
     toJSON(): GeometryPartProps;
 }
 
+// @alpha
+export const getSchemaJsonFromIModel: (iModel: IModelDb) => (schemaName: string) => string | undefined;
```

Comment:
nit (resolve at will), composing functions that return functions I think are best named like:
`makeSchemaJsonGetterFromIModel` where it indicates the construction of a noun.

Might also want to rename `GetSchemaJson` in ecschema-metadata to `SchemaJsonGetter` if you go with this.

Reply:
Thanks, @MichaelBelousov I was having trouble naming these things.  @pmconne do we have any standard for this?  Mikes seems reasonable to me.  If you agree I will add it to our coding guidelines markdown.

Reply:
Well one guideline we don't currently have is: if your return type is a function, declare a type alias for that function type, e.g., `type GetSchemaJsonStringFromName = (schemaName: string) => string` - the PascalCased version of the name you'd give to a function of this type.
Often you can make the type name into a noun based on what the function does, e.g. RpcRequestSupplier, ProgressCallback, ClientDiagnosticsHandler, TileTreeProcessor, etc.
Other times you give up and call it SomethingFunction.
In this case maybe you decide you like SchemaJsonStringFromNameGetter (?!)
Without a type alias your variable declaration is a chore to parse.

We don't have a convention for naming  functions that return functions. @MichaelBelousov's suggestion seems fine.

I don't like that your variable name claims to return (a function that returns) JSON but actually returns a string.

Reply:
I like `SchemaJsonStringGetter`

Reply:
I took your feedback, combined it with the other refactors and renamed things.  

The core method is now `makeSchemaPropsGetterFromIModel`, the type for the getter is `SchemaPropsGetter` and everything returns the `SchemaProps` interface.

I will create another PR to update our coding guidelines